nntrainer/layers/cl_layers/rmsnorm_layer_cl.cpp

```diff
@@ -0,0 +1,399 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 Debadri Samaddar <s.debadri@samsung.com>
+ *
+ * @file        fc_layer_cl.cpp
+ * @date        7 May 2020
+ * @brief       This is Fully Connected Layer Class for Neural Network with OpenCl
+ * implementation
+ * @see         https://github.com/nnstreamer/nntrainer
+ * @author      Debadri Samaddar <s.debadri@samsung.com>
+ * @bug         No known bugs except for NYI items
+ *
+ */
+
+#include <common_properties.h>
+#include <rmsnorm_layer_cl.h>
+#include <layer_context.h>
+#include <lazy_tensor.h>
+#include <nntrainer_error.h>
+#include <node_exporter.h>
+#include <util_func.h>
+#include <stdio.h>
+
+std::string rmsnorm_cl_kernel_fp16_ =
+ R"(
+    #pragma OPENCL EXTENSION cl_khr_fp16 : enable
+    __kernel void rmsnorm_cl_fp16(
+    __global const half *input,  // Input tensor
+    __global half *output,    // Output tensor
+    __global const half *alpha,  // Alpha values (one for each channel)
+    half epsilon,
+    int B,                  // Number of batches
+    int C,                  // Number of channels
+    int H,                  // Height of feature map
+    int W                   // Width of feature map
+) {
+    int global_id = get_global_id(0);  // Get the global work item index
+
+    // Compute the corresponding batch, height, and channel indices
+    int n = global_id / (H * C);       // Batch index
+    int hc = global_id % (H * C);      // Combined height and channel index
+    int h = hc / C;                    // Height index
```

Comment:
quick question!
Why do you use `n*h*c` instead of `n*c*h` in id finding?

Reply:
Hello Ms,Yang, The above calculation is just to get the n,c,h values . The index calculation is done as below
int index = ((n * C + c) * H + h) * W;

So irrespective of n*c*h or n*h*c both will give the correct index.

Reply:
Yes, I understood what you explained.
I understood these code is just calculating n, h, c values from your group format.
However, the way to compute index from `global_id` follows  `index = ((n * H + h) * C + c) * W`; the way you made work_groups seem to work like that.
I just wonder whether it is natural to find `n, h, c` format even your index follows `n*c*h`.